src/azul/portal_service.py

```diff
@@ -197,7 +201,16 @@ def _read_db(self, version: str) -> JSONs:
                                               Key=self.object_key,
                                               VersionId=version)
         except self.client.exceptions.NoSuchKey:
+            # We hypothesis that NoSuchKey is raised when the object has been
+            # created but has not materializied yet.
             raise NoSuchObjectVersion(version)
+        except ClientError as e:
+            # We hyposthesis that NoSuchVersion is raised when the object has
+            # been overwritten but the overwrite has not materilaized yet.
```

Comment:
```suggestion
            # We hypothesize that when S3 was only eventually consistent, NoSuchVersion would have been raised when the object had
            # been overwritten but the overwrite hadn't materialized yet.
```

Please be more meticulous with your spelling.